src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Future.cs

```diff
@@ -534,6 +534,33 @@ public new ConfiguredTaskAwaitable<TResult> ConfigureAwait(bool continueOnCaptur
 
         #endregion
 
+        #region WaitAsync methods
+        /// <summary>Gets a <see cref="Task{TResult}"/> that will complete when this <see cref="Task{TResult}"/> completes or when the specified <see cref="CancellationToken"/> has cancellation requested.</summary>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for a cancellation request.</param>
+        /// <returns>The <see cref="Task{TResult}"/> representing the asynchronous wait.  It may or may not be the same instance as the current instance.</returns>
+        public new Task<TResult> WaitAsync(CancellationToken cancellationToken) =>
+            WaitAsync(Timeout.UnsignedInfinite, cancellationToken);
+
+        /// <summary>Gets a <see cref="Task{TResult}"/> that will complete when this <see cref="Task{TResult}"/> completes or when the specified timeout expires.</summary>
+        /// <param name="timeout">The timeout after which the <see cref="Task"/> should be faulted with a <see cref="TimeoutException"/> if it hasn't otherwise completed.</param>
+        /// <returns>The <see cref="Task{TResult}"/> representing the asynchronous wait.  It may or may not be the same instance as the current instance.</returns>
+        public new Task<TResult> WaitAsync(TimeSpan timeout) =>
+            WaitAsync(ValidateTimeout(timeout, ExceptionArgument.timeout), default);
+
+        /// <summary>Gets a <see cref="Task{TResult}"/> that will complete when this <see cref="Task{TResult}"/> completes, when the specified timeout expires, or when the specified <see cref="CancellationToken"/> has cancellation requested.</summary>
+        /// <param name="timeout">The timeout after which the <see cref="Task"/> should be faulted with a <see cref="TimeoutException"/> if it hasn't otherwise completed.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for a cancellation request.</param>
+        /// <returns>The <see cref="Task{TResult}"/> representing the asynchronous wait.  It may or may not be the same instance as the current instance.</returns>
+        public new Task<TResult> WaitAsync(TimeSpan timeout, CancellationToken cancellationToken) =>
+            WaitAsync(ValidateTimeout(timeout, ExceptionArgument.timeout), cancellationToken);
+
+        internal new Task<TResult> WaitAsync(uint millisecondsTimeout, CancellationToken cancellationToken) =>
+            IsCompleted || (!cancellationToken.CanBeCanceled && millisecondsTimeout == Timeout.UnsignedInfinite) ? this :
+            cancellationToken.IsCancellationRequested ? FromCanceled<TResult>(cancellationToken) :
+            millisecondsTimeout == 0 ? FromException<TResult>(new TimeoutException()) :
+            new CancellationPromise<TResult>(this, millisecondsTimeout, cancellationToken);
```

Comment:
this is hard to read, please use `if else` blocks or change indentation
```suggestion
            IsCompleted || (!cancellationToken.CanBeCanceled && millisecondsTimeout == Timeout.UnsignedInfinite) 
                ? this
                : cancellationToken.IsCancellationRequested 
                    ? FromCanceled<TResult>(cancellationToken)
                    : millisecondsTimeout == 0 
                        ? FromException<TResult>(new TimeoutException())
                        : new CancellationPromise<TResult>(this, millisecondsTimeout, cancellationToken);
```

Reply:
It's funny how two people can look at two different versions of the same code and disagree on which one is harder to read ;-)

I can change it to if blocks.

Reply:
For me the compact layout looks very clear - each line has one conditional result.

Reply:
Me, too :smile:  I find the:
```
If ? Then :
ElseIf ? Then :
ElseIf ? Then :
ElseIf ? Then :
Else;
```
structure very natural to read.  But, to each their own.